edb/pgsql/compiler/relgen.py

```diff
@@ -714,6 +714,11 @@ def process_set_as_link_property_ref(
         else:
             link_prefix = ir_source
 
+        # If this is a WITH bound link, dive through a bit to get to it.
```

Comment:
We should disallow this.  There is no semantic "persistence" of the fact that you've traversed the link in a `WITH` expression.  Past that point it's just a set.